tests/test_api_events_traversal_check.py

```
@@ -0,0 +1,35 @@
+
+import pytest
+from fastapi import HTTPException, Request
+from swarm.api.routes import events
+
+
+@pytest.mark.anyio
+async def test_events_path_traversal():
+    """Test that events endpoints validate run_id against path traversal."""
+
+    # Mock runs root
+    runs_root = Path("/tmp/runs")
```

Ruff (0.14.14)

[error] Probable insecure usage of temporary file or directory: "/tmp/runs"

(S108)